proposals/0000-allow-constraint-type-synonyms-in-deriving-heads.md

```diff
@@ -0,0 +1,89 @@
+---
+author: Grigorii Gerasev
+date-accepted: ""
+ticket-url: ""
+implemented: ""
+---
+
+This proposal is [discussed at this pull request](https://github.com/ghc-proposals/ghc-proposals/pull/597).
+
+# Proposal title
+
+Allow constraints type synonyms in deriving heads again.
+
+## Motivation
+
+Most of Haskell projects do a lot of `stock` and `anyclass` deriving
+for repeating classes. Most of time there is some default list
+of classes to be defived.
+
+Making type synonym for that seems like the obvious thing to do,
+jet it does not work.
+
+It worked at some moment for any instances,
+but was disabled due to wrong behaviour in some cases:
+https://gitlab.haskell.org/ghc/ghc/-/issues/13267
```

Comment:
Thanks for asking a review from me.
But I'm not a part of Committee, I'm not a contributor to GHC or famous Hackage library. My thoughts are just my thoughts.

I like your proposal. I wish even more: use type synonyms in instance heads
```haskell
type ReadShow x = (Show x, Read x)

data MyData

instance ReadShow MyData where
    show a = ...
    read a = ...

-- same as 
instance (Show MyData, Read MyData) where
    show a = ...
    read a = ...
```